### ISSUE TITLE
Can the coordinate system change between camera poses and the mesh? [question]

### ISSUE THREAD

**Issue by Sixela963** [alicevision/Meshroom#2128]
**Describe the problem**
I want to generate a mesh and then use it, along with the images and their known associated camera poses, to do some raycasting among other things. This means that I need to know the relationship between the camera poses and the mesh. I have made a Matlab program, which could do just that. I fetch the cameras.sfm file, extract the pose of the camera associated to the image, compute its local X-Y-Z axes from that and can later do some raycasting on the mesh imported as an STL. As far as I could tell, it worked. However, when trying again with a different photogrammetry, the relationship between the camera point cloud and the STL mesh is no longer the same as in Meshroom. I cannot find anything about this coordinate change.

**Screenshots**
![image](https://github.com/alicevision/Meshroom/assets/114775005/40ce809d-0e4c-43da-ba39-34a029a83802)
![image](https://github.com/alicevision/Meshroom/assets/114775005/72d52930-5333-4b0a-bdb3-d39bf37cf1f3)
Some views of the meshroom project.

![image](https://github.com/alicevision/Meshroom/assets/114775005/fd5f05c7-ea33-4021-a33b-c481a5da2e51)
![image](https://github.com/alicevision/Meshroom/assets/114775005/97b1e008-b147-4c6d-a57b-ff1e735cfc30)
Some views of the computed coordinates in Matlab. X-Y-Z are Red-Green-Blue arrows for each point. As you can see, the cameras are apparently pointing away. Some poses also clip through the mesh.

**Dataset**
Here is an example of the poses I extract.

> 
        {
            "poseId": "53409765",
            "pose": {
                "transform": {
                    "rotation": [
                        "0.99660131807354713",
                        "-0.013065253916911313",
                        "-0.081333338515981166",
                        "-0.001510720754120027",
                        "0.98427724023317409",
                        "-0.17662398501270901",
                        "0.082362191185468239",
                        "0.17614656822956617",
                        "0.98091225701592455"
                    ],
                    "center": [
                        "0.66185410420102864",
                        "0.55171417498007047",
                        "-0.63862664992446128"
                    ]
                },
                "locked": "0"
            }
        },

The resulting matlab variables are:

> center =
    0.6619
    0.5517
   -0.6386
rotMatrix =
    0.9966   -0.0015    0.0824
   -0.0131    0.9843    0.1761
   -0.0813   -0.1766    0.9809

The XYZ vectors for each camera are computed as such:
`camXVector = inv(rotMatrix) * [1;0;0];
camYVector = inv(rotMatrix) * [0;1;0];
camZVector = inv(rotMatrix) * [0;0;1];`

Any help would be greatly appreciated.

**Desktop:**
 - OS: win 10
 - Meshroom version: 2023.2.0

**Comment by simogasp** [comment]
> However, What I then do not understand, is why this only happened in one project and not another?

I think the reference frame convention has been changed between two releases, so if you used different versions, it could explain it. Maybe @servantftechnicolor has some better insight.

**Comment by Sixela963** [comment]
> > However, What I then do not understand, is why this only happened in one project and not another?
> 
> I think the reference frame convention has been changed between two releases, so if you used different versions, it could explain it. Maybe @servantftechnicolor has some better insight.

I do not believe that to be the problem, as those two project where done within a few days at most, and I didn't reinstall or update Meshroom inbetween. I am not aware of any auto-update either.
